pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -68,6 +73,23 @@ func (cloud *CloudProvider) GetDiskOrError(ctx context.Context, volumeZone, volu
 	return disk, nil
 }
 
+func (cloud *CloudProvider) GetSnapshotOrError(ctx context.Context, snapshotName string) (*compute.Snapshot, error) {
+	svc := cloud.service
+	project := cloud.project
+	glog.Infof("Getting snapshot %v", snapshotName)
+	snapshot, err := svc.Snapshots.Get(project, snapshotName).Context(ctx).Do()
+	if err != nil {
+		if IsGCEError(err, "notFound") {
+			return nil, status.Error(codes.NotFound, fmt.Sprintf("snapshot %v does not exist", snapshotName))
```

Comment:
only return error codes at the top level rpc call. Just bubble up the GCE errors here.

(I know currently other functions have the error handling inside the controller functions but those are wrong and I'm currently changing them)